apps/wellknown/views/application.py

```diff
@@ -13,6 +16,44 @@ class ApplicationListPagination(PageNumberPagination):
     max_page_size = 10000
 
 
+class LabelSlugsValidator:
+    def __call__(self, value):
+        if value:
+            slugs = [slug.strip() for slug in value.split(',')]
+            for slug in slugs:
+                if not ApplicationLabel.objects.exclude(slug__in=settings.APP_LIST_EXCLUDE).filter(slug=slug).exists():
+                    raise ValidationError('Invalid slug name for label parameter:  %s' % (slug), code=400)
+
+
+class LabelSlugsField(forms.CharField):
+    def __init__(self, **kwargs):
+        super().__init__(validators=[LabelSlugsValidator()], **kwargs)
+
+
+class LabelSlugsFilter(Filter):
+    def __init__(self, **kwargs):
+        self.field_class = LabelSlugsField
+        super().__init__(**kwargs)
+
+    def filter(self, qs, value):
+        if value:
+            result_qs = Application.objects.none()
+            slugs = [slug.strip() for slug in value.split(',')]
+            for slug in slugs:
+                result_qs = result_qs | super().filter(qs, slug)
+        else:
+            result_qs = super().filter(qs, value)
+        return result_qs.distinct()
+
+
+class ApplicationListFilter(FilterSet):
+    label = LabelSlugsFilter(field_name="applicationlabel__slug")
```

Comment:
Can this be written such that it uses already made components and describes the behavior at this level.

e.g. `label = filters.BaseInFilter(field_name="applicationlabel__slug", validators=[LabelSlugsValidator])`

Reply:
this comment may make many of the other ones void if it works.